File: tiq-e/strings/3/main.go
package main

import (
	"fmt"
	"time"
)

func firstUniqChar(s string) int {
	m := make(map[int32]int)
	for _, c := range s {
		if v, ok := m[c]; ok {
			m[c] = v + 1
		} else {
			m[c] = 1
		}
	}
	for i, c := range s {
		if m[c] == 1 {
			return i
		}
	}
	return -1
}

func main() {
	s := "loveleetcode"

	var start = time.Now()
	var result = firstUniqChar(s)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/strings/3/main.py
from time import perf_counter


class Solution:
    def firstUniqChar(self, s: str) -> int:
        m = {}
        for i in s:
            m[i] = m.get(i, 0) + 1
            print(i)
        for i in range(len(s)):
            if m[s[i]] == 1:
                return i
        return -1


s = "loveleetcode"

start = perf_counter()
result = Solution().firstUniqChar(s)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))
